### PR TITLE
agau - Technical Training

### PR DESCRIPTION
**Pull request by agau-odoo**

### CODE REVIEW

File: estate/views/estate_property_views.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+                <field name="expected_price" />
+                <field name="bedrooms" />
+                <field name="living_area" />
+                <field name="facades" />
+                <separator />
+                <filter string="Available Properties" name="available"
+                    domain="[('state', 'in', ['new', 'offer_received'])]" />
+                <group expand="1" string="Group By">
+                    <filter string="Post Code" name="postcode"
+                        context="{'group_by':'postcode','residual_visible':True}" />
```
Comment (vava-odoo) [inline estate/views/estate_property_views.xml:103]: I don't know where you found that, but it seems unused (and useless?) in recent versions of Odoo
```suggestion
                        context="{'group_by':'postcode'}" />
```

File: estate/views/estate_property_views.xml [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    <record id="estate_property_view_search" model="ir.ui.view">
+        <field name="name">estate.property.form</field>
+        <field name="model">estate.property</field>
+        <field name="arch" type="xml">
+            <search string="Properties">
+                <field name="name" string="Title" />
+                <field name="postcode" />
+                <field name="expected_price" />
+                <field name="bedrooms" />
+                <field name="living_area" />
```
Comment (vava-odoo) [inline estate/views/estate_property_views.xml:96]: Looks like you forgot the `filter_domain`

**Review by vava-odoo — COMMENTED** [review]
Nice work!

File: estate/views/estate_property_tag_views.xml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+                    <div class="oe_title">
+                        <div class="oe_edit_only">
+                            <label for="name" />
+                        </div>
+                        <h1 class="mb32">
+                            <field name="name" class="mb16" />
+                        </h1>
+                    </div>
+                </sheet>
+            </form>
```
[the remark this answers is not in the document]
Comment (agau-odoo) [reply estate/views/estate_property_tag_views.xml:24]: My bad, I must've misread the spec. I'll remove it

File: awesome_dashboard/static/src/statistics_service.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        }
+
+        return {
+            state,
+            async loadData() {
+                await loadStatistics();
+                startInterval();
+            },
+        };
+    }
```
Comment (davidmonnom) [inline awesome_dashboard/static/src/statistics_service.js:34]:
```suggestion
export const statisticsService = {
    dependencies: ["rpc"],
    async start(env, { rpc }) {
        const state = reactive({
            orders_by_size: {},
        });

        async function loadStatistics() {
            const statistics = await rpc("/awesome_dashboard/statistics");
            Object.assign(state, statistics);
        }

        setInterval(loadStatistics, TEN_MINUTES);
        loadStatistics();

        return state;
    },
};
```

Comment (davidmonnom) [reply awesome_dashboard/static/src/statistics_service.js:34]: Adaptations may need to be made elsewhere

**Review by davidmonnom — COMMENTED** [review]
Good job! :+1: 

Here are a few comments

File: awesome_dashboard/static/src/dashboard/dashboard.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+        this.statisticsService = useService("awesome_dashboard.statistics");
+        this.dashboardConfigService = useDashboardConfig();
+        this.openDialog = useOwnedDialogs();
+
+        this.statistics = useState(this.statisticsService.state);
+        this.disabledItems = this.dashboardConfigService.disabledItems;
+        this.items = registry.category("awesome_dashboard").getAll();
+
+        onWillStart(async () => {
+            await this.statisticsService.loadData();
```
[the remark this answers is not in the document]
Comment (agau-odoo) [reply awesome_dashboard/static/src/dashboard/dashboard.js:28]: I think I got confused with the wording of the tutorial
[Part 7](https://www.odoo.com/documentation/17.0/developer/tutorials/discover_js_framework/02_build_a_dashboard.html#real-life-update):
> [...]if the dashboard is currently being displayed, it should be updated immediately.[...]

However, I still like the idea of lazy-loading the data. If a user never navigates to our dashboard app, we don't need to make unnecessary rpc calls.

File: awesome_dashboard/static/src/dashboard_config/dashboard_config_store.js [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+        return this._disabledItemsState;
+    }
+
+    set disabledItems(value) {
+        this._disabledItemsState.splice(0);
+        this._disabledItemsState.push(...value);
+    }
+
+    async save() {
+        await this.userService.setUserSettings('dashboard_settings', JSON.stringify(this.disabledItems));
```
[the remark this answers is not in the document]
Comment (agau-odoo) [reply awesome_dashboard/static/src/dashboard_config/dashboard_config_store.js:37]: I agree, but tutorial [part 12](https://www.odoo.com/documentation/17.0/developer/tutorials/discover_js_framework/02_build_a_dashboard.html#going-further) says:
> Save the content of the dashboard in a user setting on the server!

File: awesome_dashboard/static/src/pie_chart/pie_chart.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        options: {type: Object, optional: true},
+    };
+
+    setup() {
+        this.canvasRef = useRef('pie-canvas');
+
+        onWillStart(async () => {
+            await loadBundle("web.chartjs_lib");
+        });
+        useEffect(() => this.renderChart());
```
[the remark this answers is not in the document]
Comment (agau-odoo) [reply awesome_dashboard/static/src/pie_chart/pie_chart.js:19]: This change would not re-render the graph when the data changes.
In practice, I would use most likely use the GraphRenderer component that does this: https://github.com/odoo/odoo/blob/db10215dbd9a9a7472cf7877c2815d317d1e2eff/addons/web/static/src/views/graph/graph_renderer.js#L64

File: awesome_dashboard/static/src/dashboard_config/dashboard_config_store.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (davidmonnom) [reply awesome_dashboard/static/src/dashboard_config/dashboard_config_store.js:37]: Ah I  didn't know, perfect then

**Comment by agau-odoo** [comment]
Note: I squashed my commits to get the CI back, but I still have a copy of the original commit history if needed.

**Pull request closed without merging**